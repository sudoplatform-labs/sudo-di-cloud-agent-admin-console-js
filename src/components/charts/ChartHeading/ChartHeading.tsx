import React from 'react';
import styled from 'styled-components';
import { theme } from '../../../theme';

export const Heading = styled.h3`
  margin: 0;
  font-size: 14px;
`;

export const Description = styled.p`
  color: ${theme.greys.raven};
  font-size: 12px;
`;

interface Props {
  heading?: string;
  subHeading?: string;
}

export const ChartHeading: React.FC<Props> = (props) => {
  return (
    <>
      {props.heading && <Heading>{props.heading}</Heading>}
      {props.subHeading && <Description>{props.subHeading}</Description>}
    </>
  );
};
