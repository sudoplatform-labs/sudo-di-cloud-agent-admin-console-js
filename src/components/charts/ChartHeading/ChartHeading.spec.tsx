import { shallow } from 'enzyme';
import React from 'react';
import { ChartHeading } from './ChartHeading';

describe('ChartHeading', () => {
  it('should render correctly', () => {
    const component = shallow(
      <ChartHeading
        heading="I feel good"
        subHeading="I feel great, I feel wonderful"
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
