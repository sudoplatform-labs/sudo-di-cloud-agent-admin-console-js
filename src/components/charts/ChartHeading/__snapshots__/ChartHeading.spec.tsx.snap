// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartHeading should render correctly 1`] = `
<Fragment>
  <styled.h3>
    I feel good
  </styled.h3>
  <styled.p>
    I feel great, I feel wonderful
  </styled.p>
</Fragment>
`;
